pkg/generate/app/pipeline.go

```diff
@@ -341,21 +342,40 @@ func AddServices(objects Objects, firstPortOnly bool) Objects {
 	for _, o := range objects {
 		switch t := o.(type) {
 		case *deploy.DeploymentConfig:
-			ports := UniqueContainerToServicePorts(AllContainerPorts(t.Spec.Template.Spec.Containers...))
-			if len(ports) == 0 {
-				continue
+			svc := addServiceInternal(t.Spec.Template.Spec.Containers, t.ObjectMeta, t.Spec.Selector, firstPortOnly)
+			if svc != nil {
+				svcs = append(svcs, svc)
 			}
-			if firstPortOnly {
-				ports = ports[:1]
+		case *extensions.DaemonSet:
+			// Daemonsets have defaulting logic, if the selector is not set it will default to
+			// the labels on the pod template
+			selector := t.Spec.Template.Labels
+			if t.Spec.Selector != nil {
+				selector = t.Spec.Selector.MatchLabels
```

Comment:
Right, (in deployments/deploymentconfigs and I imagine it's the same for daemonsets) you can have a nil selector and it will be defaulted to the podtemplate labels.